src/nvim/main.c

```diff
@@ -905,12 +905,7 @@ static void command_line_scan(mparm_T *parmp)
           if (STRICMP(argv[0] + argv_idx, "help") == 0)
             usage();
           else if (STRICMP(argv[0] + argv_idx, "version") == 0) {
-            Columns = 80;                 /* need to init Columns */
-            info_message = TRUE;           /* use mch_msg(), not mch_errmsg() */
-            list_version();
-            msg_putchar('\n');
-            msg_didout = FALSE;
-            mch_exit(0);
+            version();
```

Comment:
It would be better to leave the explicit mch_exit(0) in both cases, instead of hiding it in version() and having to explain what's going on.